nengo/cache.py

```diff
@@ -321,6 +321,11 @@ def shrink(self, limit=None):
             Maximum size of the cache in bytes.
         """
         if self.readonly:
+            warnings.warn("Cannot shrink a readonly cache.")
```

Comment:
Sounds good to me. Not sure if I would go with `logger.info` or `logger.debug`, so either of those is fine with me. (Cache misses/hits are `debug`.)